weld/jpa/src/main/java/org/jboss/as/weld/services/bootstrap/WeldJpaInjectionServices.java

```diff
@@ -89,8 +101,12 @@ public WeldJpaInjectionServices(DeploymentUnit deploymentUnit) {
         //now we have the service controller, as this method is only called at runtime the service should
         //always be up
         final PersistenceUnitServiceImpl persistenceUnitService = (PersistenceUnitServiceImpl) serviceController.getValue();
```

Comment:
The enhancement here is to return a LazyEntityManagerFactory if we detect that the persistence unit EntityManagerFactory is still null (meaning it hasn't started yet).